--- a/Makefile.in
+++ b/Makefile.in
@@ -173,9 +173,11 @@ PSRCS:=$(foreach s, $(SRCS), $(PACKAGE_N
 PGSRCS:=$(foreach s, $(GENERATED_SRCS), $(PACKAGE_NAME)-$(PACKAGE_VERSION)/$(s))
 
 MAN_PAGES:=$(KEXEC_MANPAGE) $(VMCORE_DMESG_MANPAGE)
+MAN_PAGES+=$(MANDIR)/man8/coldreboot.8
 BINARIES_i386:=$(KEXEC_TEST)
 BINARIES_x86_64:=$(KEXEC_TEST)
 BINARIES:=$(KEXEC) $(VMCORE_DMESG) $(BINARIES_$(ARCH))
+SCRIPTS:=$(SBINDIR)/coldreboot
 
 UNINSTALL_KDUMP = $(sbindir)/kdump
 UNINSTALL_KDUMP_MANPAGE = $(mandir)/man8/kdump.8
@@ -184,7 +186,7 @@ UNINSTALL_KEXEC_MANPAGE = $(mandir)/man8
 UNINSTALL_VMCORE_DMESG = $(sbindir)/vmcore-dmesg
 UNINSTALL_VMCORE_DMESG_MANPAGE = $(mandir)/man8/vmcore-dmesg.8
 
-TARGETS:=$(BINARIES) $(MAN_PAGES)
+TARGETS:=$(BINARIES) $(MAN_PAGES) $(SCRIPTS)
 targets: $(TARGETS)
 
 UNINSTALL_TARGETS:=$(UNINSTALL_KDUMP) $(UNINSTALL_KDUMP_MANPAGE) \
--- a/kexec/Makefile
+++ b/kexec/Makefile
@@ -103,6 +103,8 @@ clean += $(KEXEC_OBJS) $(KEXEC_DEPS) $(K
 
 KEXEC = $(SBINDIR)/kexec
 KEXEC_MANPAGE = $(MANDIR)/man8/kexec.8
+KEXEC_COLDREBOOT=$(SBINDIR)/coldreboot
+KEXEC_COLDREBOOT_MANPAGE=$(MANDIR)/man8/coldreboot.8
 
 -include $(KEXEC_DEPS)
 
@@ -117,6 +119,16 @@ kexec/fs2dt.o: CPPFLAGS+=$($(ARCH)_FS2DT
 $(KEXEC_MANPAGE): kexec/kexec.8
 	@$(MKDIR) -p     $(MANDIR)/man8
 	cp $^ $(KEXEC_MANPAGE)
+
+$(KEXEC_COLDREBOOT): kexec/coldreboot
+	@$(MKDIR) -p     $(@D)
+	cp kexec/coldreboot $(KEXEC_COLDREBOOT)
+	/bin/chmod 755 $(KEXEC_COLDREBOOT)
+
+$(KEXEC_COLDREBOOT_MANPAGE): kexec/coldreboot.8
+	@$(MKDIR) -p     $(MANDIR)/man8
+	cp kexec/coldreboot.8 $(KEXEC_COLDREBOOT_MANPAGE)
+
 echo::
 	@echo "KEXEC_SRCS $(KEXEC_SRCS)"
 	@echo "KEXEC_DEPS $(KEXEC_DEPS)"
--- /dev/null
+++ b/kexec/coldreboot
@@ -0,0 +1,6 @@
+#!/bin/sh
+NOKEXECFILE=/no-kexec-reboot
+
+/bin/rm -f $NOKEXECFILE
+touch $NOKEXECFILE
+/sbin/reboot $*
--- /dev/null
+++ b/kexec/coldreboot.8
@@ -0,0 +1,25 @@
+.\" Process this file with
+.\" groff -man -Tascii coldreboot.8
+.\"
+.TH kexec 8 "April 2006" Linux "User Manuals"
+.SH NAME
+coldreboot \- Force a cold reboot bypassing kexec-tools default
+.SH SYNOPSIS
+.B /sbin/coldreboot
+.B [reboot options]
+
+.SH DESCRIPTION
+.B coldreboot
+is a script that forces a cold reboot regardless of whether kexec is enabled
+or not in /etc/default/kexec.
+.B coldreboot
+takes the same arguments as /sbin/reboot and passes them on to /sbin/reboot
+later.
+
+.SH NOTES
+.PP
+.B coldreboot
+prevents a kexec kernel from being loaded automatically and thus
+forces a cold reboot. If the user loads a kexec kernel by hand,
+.B coldreboot
+will still result in a kexec reboot.
