--- a/kexec/arch/i386/kexec-elf-x86.c
+++ b/kexec/arch/i386/kexec-elf-x86.c
@@ -91,7 +91,7 @@ int elf_x86_load(int argc, char **argv,
 	char *command_line = NULL, *modified_cmdline = NULL;
 	const char *append = NULL;
 	char *tmp_cmdline = NULL;
-	char *error_msg = NULL;
+	int error_code = 0;
 	int result;
 	int command_line_len;
 	int modified_cmdline_len;
@@ -221,7 +221,7 @@ int elf_x86_load(int argc, char **argv,
 		elf_rel_set_symbol(&info->rhdr, "entry32_regs", &regs, sizeof(regs));
 
 		if (ramdisk) {
-			error_msg = "Ramdisks not supported with generic elf arguments";
+			error_code = 1;
 			goto out;
 		}
 	}
@@ -292,13 +292,20 @@ int elf_x86_load(int argc, char **argv,
 		elf_rel_set_symbol(&info->rhdr, "entry32_regs", &regs, sizeof(regs));
 	}
 	else {
-		error_msg = "Unknown argument style\n";
+		error_code = 2;
 	}
 
 out:
 	free(command_line);
 	free(modified_cmdline);
-	if (error_msg)
-		die(error_msg);
+	switch (error_code) {
+		case 1:
+			die("Ramdisks not supported with generic elf arguments");
+		case 2:
+			die("Unknown argument style\n");
+
+		default:
+			break;
+	}
 	return result;
 }
--- a/kexec/arch/x86_64/kexec-elf-x86_64.c
+++ b/kexec/arch/x86_64/kexec-elf-x86_64.c
@@ -99,7 +99,7 @@ int elf_x86_64_load(int argc, char **arg
 #define ARG_STYLE_NONE  2
 	int opt;
 	int result = 0;
-	char *error_msg = NULL;
+	int error_code = 0;
 
 	/* See options.h and add any new options there too! */
 	static const struct option options[] = {
@@ -207,7 +207,7 @@ int elf_x86_64_load(int argc, char **arg
 		elf_rel_set_symbol(&info->rhdr, "entry64_regs", &regs, sizeof(regs));
 
 		if (ramdisk) {
-			error_msg = "Ramdisks not supported with generic elf arguments";
+			error_code = 1;
 			goto out;
 		}
 	}
@@ -269,13 +269,21 @@ int elf_x86_64_load(int argc, char **arg
 		elf_rel_set_symbol(&info->rhdr, "entry64_regs", &regs, sizeof(regs));
 	}
 	else {
-		error_msg = "Unknown argument style\n";
+		error_code = 2;
 	}
 
 out:
 	free(command_line);
 	free(modified_cmdline);
-	if (error_msg)
-		die(error_msg);
+	switch (error_code) {
+		case 1:
+			die("Ramdisks not supported with generic elf arguments");
+
+		case 2:
+			die("Unknown argument style\n");
+
+		default:
+			break;
+	}
 	return result;
 }
