From: Khalid Aziz <khalid@debian.org>
Date: Fri, 30 Apr 2021 11:17:32 -0700
Subject: distclean_fix

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index ece6206..b3bdb6f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -255,7 +255,7 @@ clean:
 distclean: dist-clean
 
 dist-clean: clean
-	$(RM) -f config.log config.status config.cache Makefile include/config.h
+	$(RM) -f config.log config.status config.cache Makefile include/config.h include/stamp-h*
 	$(RM) -f include/config.h.in configure $(SPEC)
 	$(RM) -rf autom4te.cache
 
