From: Khalid Aziz <khalid@debian.org>
Date: Fri, 30 Apr 2021 11:17:32 -0700
Subject: static_binary_warning

===================================================================
---
 kexec_test/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kexec_test/Makefile b/kexec_test/Makefile
index fec6210..82b7e87 100644
--- a/kexec_test/Makefile
+++ b/kexec_test/Makefile
@@ -20,7 +20,7 @@ ifeq ($(BUILD_KEXEC_TEST),yes)
 KEXEC_TEST_OBJS = $(call objify, $(KEXEC_TEST_SRCS))
 KEXEC_TEST_DEPS = $(call depify, $(KEXEC_TEST_OBJS))
 
-KEXEC_TEST = $(PKGLIBDIR)/kexec_test
+KEXEC_TEST = $(PKGLIBDIR)/kexec_test.static
 
 clean += $(KEXEC_TEST_OBJS) $(KEXEC_TEST_DEPS) $(KEXEC_TEST)
 
@@ -36,5 +36,6 @@ $(KEXEC_TEST): LDFLAGS=-melf_i386 -e _start -Ttext $(RELOC)
 $(KEXEC_TEST): $(KEXEC_TEST_OBJS)
 	mkdir -p $(@D)
 	$(TARGET_LD) $(LDFLAGS) -o $@ $^
+	strip $@
 
 endif
